domino/flyte/task.py

```diff
@@ -112,16 +121,28 @@ def __init__(
 
         resolved_job_config = self._resolve_job_properties(domino_job_config)
 
-        super().__init__(
+        task = super().__init__(
             name=name,
             task_type="domino_job",
             task_config=resolved_job_config,
-            interface=Interface(),
+            interface=Interface(inputs=inputs, outputs=outputs),
             inputs=inputs,
             outputs=outputs,
             metadata=TaskMetadata(retries=0, timeout=timedelta(hours=3)),
             **kwargs,
         )
+        # the flyte init container (which downloads inputs) and flyte sidecar (which uploads outputs) require these
+        # base64 string encodings of the input/output interfaces as args to their container startup commands
+        # TODO: can detect "empty interfaces" here and omit the interface args so dont even create the init/sidecar containers if not needed
+        serialized_input_interface = task._interface.to_flyte_idl().inputs.SerializeToString()
+        serialized_input_output_interface = job._interface.to_flyte_idl().SerializeToString()
+        pipelineConfig = PipelineConfig(
+            InputInterfaceBase64=base64.b64encode(serialized_input_interface),
+            InputOutputInterfaceBase64=base64.b64encode(serialized_input_output_interface),
+        )
+        task.task_config.PipelineConfig = pipelineConfig
```

Comment:
Seems like you could create the `Interface` you use here before you call `super.__init__()` and avoid having to update the task after it is created. Not sure if it matters either way.

Also, is there a reason that one of these uses `task._interface` and the other uses `job._interface`? I'm not sure where `job` is defined here.

Reply:
to the `job` question -- it was a copy/paste messup from my other code -- will change to `task`